src/gas/GasFeeController.ts

```diff
@@ -241,10 +254,13 @@ export class GasFeeController extends BaseController<typeof name, GasFeeState> {
     fetchEthGasPriceEstimate?: typeof defaultFetchEthGasPriceEstimate;
     fetchLegacyGasPriceEstimates?: typeof defaultFetchLegacyGasPriceEstimates;
     getCurrentNetworkEIP1559Compatibility: () => Promise<boolean>;
+    getCurrentNetworkLegacyGasAPICompatibility: () => boolean;
     getCurrentAccountEIP1559Compatibility?: () => boolean;
-    getIsMainnet: () => boolean;
+    getChainId: () => `0x${string}` | number;
```

Comment:
we use `string` along the controllers like `'1'` or `'56'` instead of numbers

Reply:
just pushed a new type so it'll accept hex prefixed string, a plain number, or numeric string. It only does parseInt for hex prefixed strings.